spring-integration-core/src/main/java/org/springframework/integration/transaction/DefaultTransactionSynchronizationFactory.java

```diff
@@ -46,13 +49,16 @@ public DefaultTransactionSynchronizationFactory(TransactionSynchronizationProces
 	@Override
 	public TransactionSynchronization create(Object key) {
 		Assert.notNull(key, "'key' must not be null");
-		DefaultTransactionalResourceSynchronization synchronization = new DefaultTransactionalResourceSynchronization(key);
-		TransactionSynchronizationManager.bindResource(key, synchronization.getResourceHolder());
-		return synchronization;
+		if (!TransactionSynchronizationManager.hasResource(key)) {
```

Comment:
Well, no. I'm not agreed.

That is really the factory responsibility to bind resource to the TX or not.
And since there is no guarantee that factory will always return the same `Synchronization` instance for the same `key`, we can't unequivocally make a decision in the caller what to do.

I only see the `null` way do not create extra `Synchronization` from our default factories and do nothing afterwards.